imap/sync_support.c

```diff
@@ -2604,7 +2647,17 @@ static int sync_mailbox_compare_update(struct mailbox *mailbox,
         /* after LAST_UID, it's an append, that's OK */
         else {
             /* skip out on the first pass */
-            if (!doupdate) continue;
+            if (!doupdate) {
```

Comment:
I get the slight optimisation here (don't do it on both passes) but generally this code repeats the checks on the doupdate path as well... should this just do it both times and move the doupdate check to after?